feat:Dockable submitter

eherozhao opened the pull request:
### What was the problem/requirement? (What/Why)
We want to have a submitter with only aerender to avoid issues from adaptor.
### What was the solution? (How)
Pushed the branch with aerender only dockable submitter
### What is the impact of this change?
unblock more customers
### How was this change tested?
manual test locally by submitting jobs

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

eherozhao committed:
fix: fix the hatch linting

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
Merge branch 'aws-deadline:mainline' into mainline

eherozhao committed:
Merge branch 'aws-deadline:mainline' into mainline

eherozhao committed:
feat: dockable submitter using aerender

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore(deps): update release to fit python-semantic-release 9.12

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore(deps): update release to fix python-semantic-release 9.12 (#61)

* fix: fix the hatch linting

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

* chore(deps): update release to fit python-semantic-release 9.12

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

---------

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
Merge branch 'aws-deadline:mainline' into mainline

eherozhao committed:
remove all workflows which not useful for this branch

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore:update reg expression

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore:change back to previous reg expression

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore:remove unused code for sonarcloud check

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
chore:cleanup code

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

crowecawcaw commented on README.md:
```diff
@@ -1,117 +1,70 @@
-[![pypi](https://img.shields.io/pypi/v/deadline-cloud-for-after-effects.svg?style=flat)](https://pypi.python.org/pypi/deadline-cloud-for-after-effects)
-[![python](https://img.shields.io/pypi/pyversions/deadline-cloud-for-after-effects.svg?style=flat)](https://pypi.python.org/pypi/deadline-cloud-for-after-effects)
-[![license](https://img.shields.io/pypi/l/deadline-cloud-for-after-effects.svg?style=flat)](https://github.com/aws-deadline/deadline-cloud-for-after-effects/blob/mainline/LICENSE)
+# AWS Deadline CLoud for After Effects: Dockable Submitter
```
What's a "dockable submitter"? Also this repo could contain things besides just a submitter in the future right? The old title might have been good enough.

eherozhao replied:
That makes sense. Will update.

crowecawcaw commented on README.md:
```diff
@@ -1,117 +1,70 @@
-### Submitter Usage
+1. After Effects 24
+1. Python3.9 or higher; and
+1. Windows or a MacOS operating System
```
Nit: "... or Mac"

crowecawcaw commented on README.md:
```diff
@@ -1,117 +1,70 @@
-2. Once inside the project, navigate to the `File` Panel at the top.
-3. Under the `Scripts` sub-menu, select the `DeadlineCloudSubmitter.jsx` script to start it.
+    Windows: Select Edit > Preferences > Scripting & Expressions > select Allow Scripts To Write Files And Access Network.
+    macOS: Select After Effects > Settings > Scripting & Expressions > select Allow Scripts To Write Files And Access Network.
```
Should this be part of the install process?

eherozhao replied:
Makes sense. Will make the change.

crowecawcaw commented on dist/DeadlineCloudSubmitter_Assets/JobTemplate/template.yaml:
```diff
@@ -0,0 +1,109 @@
+      control: HIDDEN
+      label: Output File(s)
+      groupLabel: Frame Range
+    default: ~\Desktop\output
```
Will this have odd effects on Windows?

eherozhao replied:
should not. We have Artist used the tool in Windows before. Will take an action to improve this.

crowecawcaw commented on dist/DeadlineCloudSubmitter_Assets/JobTemplate/template.yaml:
```diff
@@ -0,0 +1,109 @@
+      groupLabel: Frame Range
+    default: ~\Desktop\output
+    description: The render output destination
+  - name: FrameStarts
+    type: STRING
+    userInterface:
+      control: LINE_EDIT
+      label: Starting Frames
+      groupLabel: Frame Range
+    default: 1-10
+  - name: FrameEnds
+    type: STRING
+    userInterface:
+      control: LINE_EDIT
+      label: Ending Frames
+      groupLabel: Frame Range
+    default: 1-10
```
Why are the start and stop defined by frame ranges? I'm used to seeing a single start and stop frame.

eherozhao replied:
This one works for separate frames into tasks for image sequence. I will create an Action Item to validate and improve it.

crowecawcaw commented on dist/DeadlineCloudSubmitter_Assets/JobTemplate/template.yaml:
```diff
@@ -0,0 +1,109 @@
+      label: Ending Frames
+      groupLabel: Frame Range
+    default: 1-10
+  - name: MultiFrameMaxCPU
+    type: INT
+    userInterface:
+      control: SPIN_BOX
+      label: Multi-Frame Max CPU Percentage
+      groupLabel: Source
+    description: The maximum percentage of the CPU Multi-Frame Rendering is allowed to use. Set to 0 to disable Multi-Frame Rendering. Max is 100.
+    default: 90
```
From the README, this doesn't work right? Might be better not to include it.

eherozhao replied:
Yea, sounds reasonable. I can make a change to remove everything related to this feature from this submitter now.

crowecawcaw commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+        thisObj :
+        new Window(
+            "palette",
+            "Submit Queue to AWS Deadline Cloud",
```
Could you share a screenshot of what this looks like? Do users add work to an AE queue and then submit the whole queue to Deadline? The window title seems confusing to me.

eherozhao replied:
Yea, will update the wording. I tried to make it both dockable and undockable so customers can choose the way they want. This will only show when it is undockable but the README we have and recommend is the dockable way. 
<img width="659" alt="image" src="https://github.com/user-attachments/assets/66a508cb-8c6d-4658-bcb0-e00881b6fcb8">

crowecawcaw commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+        new Window(
+            "palette",
+            "Submit Queue to AWS Deadline Cloud",
+            undefined, {
+                resizable: true,
+                closeButton: true,
+            }
```
Super nit: formatting. Would be nice to run this through prettier if it doesn't choke on the `#include` statements.

eherozhao replied:
yea, I used a formatter but not prettier since this is extendScript. It has a specific plugin for ExtendScript https://github.com/aws-deadline/deadline-cloud-for-after-effects/blob/mainline/DEVELOPMENT.md so I just format the files with this.

dependabot[bot] committed:
chore(deps): Update pytest-cov requirement from ==5.* to ==6.* (#64)

Updates the requirements on [pytest-cov](https://github.com/pytest-dev/pytest-cov) to permit the latest version.
- [Changelog](https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/pytest-dev/pytest-cov/compare/v5.0.0...v6.0.0)

---
updated-dependencies:
- dependency-name: pytest-cov
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

eherozhao committed:
Merge branch 'aws-deadline:mainline' into mainline

eherozhao committed:
chore:update the PR to address comments: remove multiframe, update wording and README

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao committed:
resolve conflict

eherozhao committed:
chore:remove requirement testing file

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

crowecawcaw reviewed (commented):
Code looks good to me. I don't have AE set up and haven't tested it manually, it'd be good if the second reviewer had more context on it.

Your strategy of merging a stripped down submitter which works and then adding back features that might be missing makes sense to me. It gives us a working submitter quicker to unblock people, and it lets us develop from a codebase that we understand. It might help to make a list of the things that will be dropped here including user-facing features and ops stuff like `CONTRIBUTING.md` and the GitHub actions so we can make sure to add them back later.

eherozhao committed:
chore:add CodeQL

Signed-off-by: Hao Zhao <36955586+eherozhao@users.noreply.github.com>

eherozhao commented:
> Code looks good to me. I don't have AE set up and haven't tested it manually, it'd be good if the second reviewer had more context on it.
> 
> Your strategy of merging a stripped down submitter which works and then adding back features that might be missing makes sense to me. It gives us a working submitter quicker to unblock people, and it lets us develop from a codebase that we understand. It might help to make a list of the things that will be dropped here including user-facing features and ops stuff like `CONTRIBUTING.md` and the GitHub actions so we can make sure to add them back later.

Thanks for the feedback. Yea, I have created internal tickets to track those action items.

eng-josh commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+
+// Setup logger
+var _scriptFileName = "OpenAeSubmitter.jsx";
+var logFileName = "aftereffects" + ".log";
```
Why not all one string? Is there a variable missing here that should have been included in the string concat?

eherozhao replied:
Will make the change.

eng-josh commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+// Setup logger
+var _scriptFileName = "OpenAeSubmitter.jsx";
+var logFileName = "aftereffects" + ".log";
+var logDirectoryPath = dcUtil.getUserDirectory() + "/.deadline/logs/submitters/";
+logDirectoryPath = dcUtil.normPath(logDirectoryPath)
```
Nit: Even though `const` technically doesn't exist in Adobe ExtendScript language, we should declare a new variable as opposed to using reassignment.

eherozhao replied:
Makes sense. I will make the update.

eng-josh reviewed (approved):
Overall, the code looks like it will run fine. Do we need to check in with Doc Writers are on any of the language for messages we may be displaying the customers?

eherozhao commented:
> Overall, the code looks like it will run fine. Do we need to check in with Doc Writers are on any of the language for messages we may be displaying the customers?

Yes, before we change the status to Stable from Alpha, I think we will need to have Doc Writers' review on our README.

npmacl commented on dist/DeadlineCloudSubmitter_Assets/JobTemplate/scripts/aerender.ps1:
```diff
@@ -0,0 +1,36 @@
+$renderarg = @("-project", "`"$project`"", "-rqindex", $rqindex, "-s", $start, "-e", $end, "-v", "ERRORS_AND_PROGRESS", "-close", "DO_NOT_SAVE_CHANGES", "-sound", "OFF")
+
+if (-Not "$outputpath".Contains(",")) {
+    $renderarg += "-output", "`"$outputpath`""
```
I would think a user could append additional arguments by making the value of outputpath something like
`path/to/output --aditional-param1 additional_value1 --additional-param2 additional_value2`. Is there anything a user could do with this we should care about?

eherozhao replied:
Not really, from customer's perspective, they should not touch this PS file. The passed in outputpath is auto detected from the submitter so the proposed case should not appear.

npmacl commented on dist/DeadlineCloudSubmitter_Assets/JobTemplate/template.yaml:
```diff
@@ -0,0 +1,100 @@
+    default: 1
+  - name: OutputFiles
+    type: PATH
+    objectType: FILE
```
The name here is plural, but the objectType of this path is `FILE` which I would think means this variable represents a single output file that would be generated. If this is just a single file, could we update the name to be singular?

eherozhao replied:
Updated to singler. We can add plural back when supporting multiple output modules.

npmacl commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+        [1.0, 0.2, 0.2],
+        1
+    );
+    errorText.text = "⚠ ERROR: Insufficient Script Permissions ⚠";
```
does ⚠ render correctly in worker agent logs?

eherozhao replied:
This will not be logged in worker agent side. This will just print out in AE window, like this.
![image](https://github.com/user-attachments/assets/c3aacca8-e4c0-4789-8d95-ccafec415139)

eherozhao committed:
fix: update the variables with no reassignment

eherozhao committed:
fix:update plural outputfiles to outputfile with no multiple output module support

Ahuge commented:
Are we able to modify the job template so that it calls the Step the name of the Comp that it is rendering?
Similar to the adaptor workflow?

New submitter
![image](https://github.com/user-attachments/assets/0454140d-2eb4-44c8-bfc3-d6af6d0670ef)


Old submitter
![image](https://github.com/user-attachments/assets/ec410bda-13d6-44c4-9c1f-10f600b462bf)

eherozhao commented:
> Are we able to modify the job template so that it calls the Step the name of the Comp that it is rendering? Similar to the adaptor workflow?
> 
> New submitter ![image](https://private-user-images.githubusercontent.com/704821/393366155-0454140d-2eb4-44c8-bfc3-d6af6d0670ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MTA1ODksIm5iZiI6MTczMzUxMDI4OSwicGF0aCI6Ii83MDQ4MjEvMzkzMzY2MTU1LTA0NTQxNDBkLTJlYjQtNDRjOC1iZmMzLWQ2YWY2ZDA2NzBlZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNlQxODM4MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xODE1M2YzYzViZDc5ZmM5YjU4NjgwN2RiMDRkN2ZhM2E0NDE5Yjg2ZDM3OWIzZmQ2OWRjYzhkNjEzZDdmZmE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CC3gha-H0ed7YvTDICQWpQW7X0z7MNKnAQ5eCyrm5pQ)
> 
> Old submitter ![image](https://private-user-images.githubusercontent.com/704821/393366363-ec410bda-13d6-44c4-9c1f-10f600b462bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MTA1ODksIm5iZiI6MTczMzUxMDI4OSwicGF0aCI6Ii83MDQ4MjEvMzkzMzY2MzYzLWVjNDEwYmRhLTEzZDYtNDRjNC05YzFmLTEwZjYwMGI0NjJiZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNlQxODM4MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZWUwNWE5ZjdkMjYyMDk3NWJjODg2ZDY4NTdkZDAwOGQzOTg3YzFjMjFiODg2MWRhYjBlYmY0ZmU3ZWMxYjJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8oScA2hQ48yHV_zb3PP7H1UuFa0w51QZctLexvkgjLM)

Thanks for the comment. Yes, I can look into it.

viknith commented on src/OpenAESubmitter.jsx:
```diff
@@ -0,0 +1,72 @@
+    return securitySetting == 1;
+}
+
+#include "UI/SubmitterUI.jsx"
```
Any reason why we should stagger the imports like this where there's logger first, then a batch of imports, then the submitter UI at the end?

eherozhao replied:
Yes, the reason is we are having a python script as a build system to combine all jsx files to a giant single file as the final submitter jsx script. In this case, this include will be replaced by the content of the file and we want the logger to be initialized before so that any logger in SubmitterUI can work. I can change the order a bit in next revision.

viknith commented on src/submission/JobTemplateHelper.jsx:
```diff
@@ -0,0 +1,154 @@
+/**
+ * Generates the basic paramterValue file for the job template
```
Nit: parameter

viknith commented on src/ui/SubmitterUI.jsx:
```diff
@@ -0,0 +1,163 @@
+
+    var submitButton = controlsGroup.add("button", undefined, "Submit");
+    submitButton.onClick = function() {
+        SubmitSelection(list.selection, separateFramesCheckbox.value ? parseInt(framesPerTaskValue.text) : 1);
```
Nit: I know list gets set because of the updateList() call further below, but should we consider a null check on list here if it's possible for list.selection to result in an error?

eherozhao replied:
Yes, we have a null check in the `SubmitSelection` method.

viknith commented on src/submission/SubmitBundle.jsx:
```diff
@@ -0,0 +1,174 @@
+    // first we must verify that our selection is valid
+    if (selection == null) {
+        adcAlert("Error: No selection");
+        return false;
```
I noticed there are empty `return` statements below but here it's `return false`. Does a return type matter? It doesn't look like it, so as a nit, we'd want to it to be consistent to minimize confusion.

eherozhao committed:
fix: 1. add try catch for aerender ps; 2. update step name to be comp name 3. minor fixes
